File: day01/puzzle.rb
# frozen_string_literal: true

input = File.read(*ARGV).split("\n\n")
group_totals = input.map { |snack_group| snack_group.split.sum(&:to_i) }

p group_totals.max # p1-> 68802

p group_totals.max(3).sum # p2-> 205370

File: day01/puzzle.js
const fs = require('fs');
const readFile = (path) => fs.readFileSync(path, { encoding: 'utf8' });

const input = readFile(process.argv[2]).slice(0, -2).split("\n\n");
const sum = (a, b) => Number(a) + Number(b);

const groupTotals = input.map((snackGroup) => {
  return snackGroup.split("\n").reduce(sum, 0);
})

console.log(Math.max(...groupTotals)); // p1

const sortedTotals = [...groupTotals].sort((a, b) => b - a);
console.log(sortedTotals.slice(0, 3).reduce(sum, 0)); // p2
